tests/lib/rules/require-atomic-updates.js

```diff
@@ -150,6 +150,18 @@ ruleTester.run("require-atomic-updates", rule, {
                 let bar = await get(foo.id);
                 foo.prop = bar.prop;
             }
+        `,
+
+        // https://github.com/eslint/eslint/issues/11954
+        `
+            let count = 0
+            let queue = []
+            async function A(...args) {
+                count += 1
+                await new Promise(resolve=>resolve())
+                count -= 1
+                return
```

Comment:
A few parts from the original snippet in #11954 aren't required to reproduce the bug, so let's remove them. That'll keep the test case as minimal and narrowly focused as possible.

```suggestion
            async function A() {
                count += 1
                await new Promise(resolve=>resolve())
                count -= 1
```